### PR TITLE
[ITG-104] Add icon of type to the task.

### PR DESCRIPTION
**Pull request by gregori0o**

### CODE REVIEW

**Comment by bkulawska** [comment]
Something's not working when I try to run the application, idk if it's my fault:
image

**Comment by gregori0o** [comment]
> Something's not working when I try to run the application, idk if it's my fault or what: image

Try re-run `docker-compose up` after you changed branch to this.

File: cabra/src/pages/components/TaskListItem.tsx [hunk 1]
```diff
@@ -76,6 +85,13 @@ export default function TaskListItem({ task }: Props) {
             </Link>
           </Card>
         </div>
+        {task.relationTypes.length > 0 && (
+          <Card tw="flex gap-2">
+            {task?.relationTypes.map((type) => (
+              <TaskTypeIcon key={type} type={type} />
+            ))}
+          </Card>
+        )}
```
Comment (koszar91) [inline cabra/src/pages/components/TaskListItem.tsx:94]: Got some remarks about this div.
It is visible conditionally, and I think it's bad. I think we should render it always, maybe with some faded empty containers to show the task is unrelated.
Also not-constant task size is vary bad.
What about this concept (it includes resizing all cards in the task a bit):
image

File: cabra/src/pages/components/TaskListItem.tsx [hunk 2]
```diff
@@ -44,8 +46,15 @@ export default function TaskListItem({ task }: Props) {
...
+  const isSubtask = task.relationTypes.includes(DirectedRelationType.Subtask);
+
   return (
-    <div tw="rounded-xl w-full bg-primary text-stone-50 shadow-2xl p-4 grid grid-cols-[1fr 30%] gap-x-4">
+    <div
+      css={[
+        tw`rounded-xl w-full text-stone-50 shadow-2xl p-4 grid grid-cols-[1fr 30%] gap-x-4`,
+        isSubtask ? tw`bg-primary bg-opacity-60` : tw`bg-primary`,
+      ]}
+    >
```
Comment (koszar91) [inline cabra/src/pages/components/TaskListItem.tsx:57]: I think coloring the subtask is not necessary. We have an icon, it's enough information.

**Review by koszar91 — CHANGES_REQUESTED** [review]
Good job, the icons are beautiful. Let's consider little redesign of the task list item, it;s getting more and more populated with data 😍

File: cabra/src/pages/components/TaskTypeIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+  return (
+    <Icon>
+      {icon}
+      <ReactTooltip id={type} place="bottom" effect="solid">
+        {message}
+      </ReactTooltip>
+    </Icon>
+  );
+}
```
Comment (john-sonz) [inline cabra/src/pages/components/TaskTypeIcon.tsx:62]: You can shorten this bit with some smart object shenanigans:
```suggestion
const TYPES_TO_ICONS = {
  [DirectedRelationType.Subtask]: {
    Icon: DuplicateIcon,
    message: "Subtask",
  },
  [DirectedRelationType.Supertask]: {
    Icon: CollectionIcon,
    message: "Has subtask",
  },
  [DirectedRelationType.Blocks]: {
    Icon: ArrowCircleDownIcon,
    message: "Blocks another task",
  },
  [DirectedRelationType.IsBlockedBy]: {
    Icon: ArrowCircleUpIcon,
    message: "Is blocked by another task",
  },
  [DirectedRelationType.Interchangable]: {
    Icon: SwitchHorizontalIcon,
    message: "Interchangeable task",
  },
};

export function TaskTypeIcon({ type }: Props) {
  const { Icon, message } = TYPES_TO_ICONS[type];

  return (
    <IconContainer>
      <Icon height={20} width={20} data-tip data-for={type} />
      <ReactTooltip id={type} place="bottom" effect="solid">
        {message}
      </ReactTooltip>
    </IconContainer>
  );
}
```

**Review by john-sonz — APPROVED** [review]
LGTM!

**Review by bkulawska — APPROVED** [review]
Looks nice <3

**Review by koszar91 — APPROVED** [review]
b e a u t i f u l  ❤

**Pull request merged**